File: app/src/main/java/io/palaima/calendar/data/DateAdapter.java
package io.palaima.calendar.data;

import android.content.ContentValues;
import android.database.Cursor;

import com.squareup.sqldelight.ColumnAdapter;

import java.util.Calendar;

public class DateAdapter implements ColumnAdapter<Calendar> {
    @Override public void marshal(ContentValues contentValues, String columnName, Calendar date) {
        contentValues.put(columnName, date.getTimeInMillis());
    }

    @Override public Calendar map(Cursor cursor, int columnIndex) {
        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(cursor.getLong(columnIndex));
        return calendar;
    }
}

File: app/src/main/java/io/palaima/calendar/data/DateAdapter.kt
package io.palaima.calendar.data

import android.content.ContentValues
import android.database.Cursor

import com.squareup.sqldelight.ColumnAdapter

import java.util.Calendar

class DateAdapter : ColumnAdapter<Calendar> {
    override fun marshal(contentValues: ContentValues, columnName: String, date: Calendar) {
        contentValues.put(columnName, date.timeInMillis)
    }

    override fun map(cursor: Cursor, columnIndex: Int): Calendar {
        val calendar = Calendar.getInstance()
        calendar.timeInMillis = cursor.getLong(columnIndex)
        return calendar
    }
}
